### ISSUE TITLE
Bard image disappears when editing entry

### ISSUE THREAD

**Issue by mikemartin** [statamic/cms#5861]
### Bug description

There seems to be a bug with the Bard fieldtype when using inline images. Entering text above an image causes the image to be replaced by the next image.

![Apr-19-2022 23-08-39](https://user-images.githubusercontent.com/414211/164011336-eaad94bf-a206-41a1-98aa-593830c30448.gif)

### How to reproduce

1. Create an entry that uses the bard fieldtype
2. Add two images
3. Save
4. Edit the entry
5. Enter text above the first image
6. See the image get replaced

### Logs

```shell
[2022-04-19 12:30:28] local.ERROR: Statamic\Assets\AssetRepository::find(): Argument #1 ($asset) must be of type string, null given, called in /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php on line 261 {"userId":"5b952bde-f01f-4caf-a960-efec848f1ceb","exception":"[object] (TypeError(code: 0): Statamic\\Assets\\AssetRepository::find(): Argument #1 ($asset) must be of type string, null given, called in /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php on line 261 at /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Assets/AssetRepository.php:33)
[stacktrace]
#0 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Support/Facades/Facade.php(261): Statamic\\Assets\\AssetRepository->find(NULL)
#1 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Fieldtypes/Assets/Assets.php(130): Illuminate\\Support\\Facades\\Facade::__callStatic('find', Array)
#2 [internal function]: Statamic\\Fieldtypes\\Assets\\Assets->Statamic\\Fieldtypes\\Assets\\{closure}(NULL, 0)
#3 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Collections/Collection.php(695): array_map(Object(Closure), Array, Array)
#4 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Fieldtypes/Assets/Assets.php(133): Illuminate\\Support\\Collection->map(Object(Closure))
#5 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Controllers/CP/Assets/FieldtypeController.php(14): Statamic\\Fieldtypes\\Assets\\Assets->getItemData(Array)
#6 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(54): Statamic\\Http\\Controllers\\CP\\Assets\\FieldtypeController->index(Object(Illuminate\\Http\\Request))
#7 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(45): Illuminate\\Routing\\Controller->callAction('index', Array)
#8 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/Route.php(262): Illuminate\\Routing\\ControllerDispatcher->dispatch(Object(Illuminate\\Routing\\Route), Object(Statamic\\Http\\Controllers\\CP\\Assets\\FieldtypeController), 'index')
#9 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/Route.php(205): Illuminate\\Routing\\Route->runController()
#10 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/Router.php(721): Illuminate\\Routing\\Route->run()
#11 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Routing\\Router->Illuminate\\Routing\\{closure}(Object(Illuminate\\Http\\Request))
#12 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/CP/CountUsers.php(18): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#13 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CP\\CountUsers->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#14 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/CP/Localize.php(19): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#15 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CP\\Localize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#16 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/CP/Authorize.php(25): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#17 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CP\\Authorize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#18 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/CP/AddToasts.php(22): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#19 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CP\\AddToasts->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#20 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/CP/AuthGuard.php(14): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#21 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CP\\AuthGuard->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#22 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/CP/ContactOutpost.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#23 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CP\\ContactOutpost->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#24 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/Middleware/SubstituteBindings.php(50): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#25 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Routing\\Middleware\\SubstituteBindings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#26 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(78): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#27 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\VerifyCsrfToken->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#28 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#29 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\View\\Middleware\\ShareErrorsFromSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#30 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(121): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#31 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(64): Illuminate\\Session\\Middleware\\StartSession->handleStatefulRequest(Object(Illuminate\\Http\\Request), Object(Illuminate\\Session\\Store), Object(Closure))
#32 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Session\\Middleware\\StartSession->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#33 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#34 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\AddQueuedCookiesToResponse->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#35 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(67): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#36 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Cookie\\Middleware\\EncryptCookies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#37 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/SwapExceptionHandler.php(19): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#38 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\SwapExceptionHandler->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#39 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#40 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/Router.php(723): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#41 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/Router.php(698): Illuminate\\Routing\\Router->runRouteWithinStack(Object(Illuminate\\Routing\\Route), Object(Illuminate\\Http\\Request))
#42 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/Router.php(662): Illuminate\\Routing\\Router->runRoute(Object(Illuminate\\Http\\Request), Object(Illuminate\\Routing\\Route))
#43 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Routing/Router.php(651): Illuminate\\Routing\\Router->dispatchToRoute(Object(Illuminate\\Http\\Request))
#44 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(167): Illuminate\\Routing\\Router->dispatch(Object(Illuminate\\Http\\Request))
#45 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(128): Illuminate\\Foundation\\Http\\Kernel->Illuminate\\Foundation\\Http\\{closure}(Object(Illuminate\\Http\\Request))
#46 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/DisableFloc.php(18): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#47 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\DisableFloc->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#48 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/CheckMultisite.php(14): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#49 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CheckMultisite->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#50 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/CheckComposerJsonScripts.php(14): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#51 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\CheckComposerJsonScripts->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#52 /Users/mike/Sites/thekeeperskitchen/vendor/statamic/cms/src/Http/Middleware/PoweredByHeader.php(19): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#53 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Statamic\\Http\\Middleware\\PoweredByHeader->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#54 /Users/mike/Sites/thekeeperskitchen/vendor/barryvdh/laravel-debugbar/src/Middleware/InjectDebugbar.php(60): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#55 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Barryvdh\\Debugbar\\Middleware\\InjectDebugbar->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#56 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#57 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ConvertEmptyStringsToNull.php(31): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#58 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ConvertEmptyStringsToNull->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#59 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php(21): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#60 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/TrimStrings.php(40): Illuminate\\Foundation\\Http\\Middleware\\TransformsRequest->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#61 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\TrimStrings->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#62 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/ValidatePostSize.php(27): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#63 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\ValidatePostSize->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#64 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/PreventRequestsDuringMaintenance.php(86): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#65 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Foundation\\Http\\Middleware\\PreventRequestsDuringMaintenance->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#66 /Users/mike/Sites/thekeeperskitchen/vendor/fruitcake/laravel-cors/src/HandleCors.php(38): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#67 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Fruitcake\\Cors\\HandleCors->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#68 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Http/Middleware/TrustProxies.php(39): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#69 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(167): Illuminate\\Http\\Middleware\\TrustProxies->handle(Object(Illuminate\\Http\\Request), Object(Closure))
#70 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(103): Illuminate\\Pipeline\\Pipeline->Illuminate\\Pipeline\\{closure}(Object(Illuminate\\Http\\Request))
#71 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(142): Illuminate\\Pipeline\\Pipeline->then(Object(Closure))
#72 /Users/mike/Sites/thekeeperskitchen/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(111): Illuminate\\Foundation\\Http\\Kernel->sendRequestThroughRouter(Object(Illuminate\\Http\\Request))
#73 /Users/mike/Sites/thekeeperskitchen/public/index.php(52): Illuminate\\Foundation\\Http\\Kernel->handle(Object(Illuminate\\Http\\Request))
#74 /Users/mike/.composer/vendor/laravel/valet/server.php(234): require('/Users/mike/Sit...')
#75 {main}
"}
```

### Versions

Statamic 3.3.7 Pro
Laravel 8.83.4
PHP 8.0.13
aerni/advanced-seo 1.0.4

### Installation

Starter Kit using via CLI

### Antlers Parser

runtime (new)

### Additional details

Using the multisimplicity starter kit.

**Comment by jasonvarga** [comment]
I can't replicate this. Can you provide a repo?

**Comment by jasonvarga** [comment]
I can't replicate this. Can you provide a repo?

**Comment by mikemartin** [comment]
Hey @jasonvarga 
I was able to replicate the AssetRepository error using the multisimplicity starter kit but not the image disappearing. 

1. statamic new mysite statamic/starter-kit-multisimplicity
2. Create new Event entry
3. Add an inline image
4. View laravel.log 

<img width="1549" alt="Screen Shot 2022-04-20 at 8 45 33 am" src="https://user-images.githubusercontent.com/414211/164113994-c627b4da-3643-4aaf-91cf-5f4f0a653fbd.png">

**Comment by jasonvarga** [comment]
The error in the log is a duplicate of #5860 which should be fixed by #5861

If you can't replicate the image disappearing, I'll close this for now but feel free to comment on here if you can reproduce it later.

**Comment by mikemartin** [comment]
Hey @jasonvarga,
The image is still disappearing after the latest update but there are no Laravel or JS errors logged. I sent you access to the private repo. Could you can pull it down and see if you can replicate it? Here's the entry where it happens:
https://github.com/mikemartin/thekeeperskitchen/blob/main/content/collections/events/2023-03-16.tuesday-tea-tunes.md
